src/manifold/src/csg_tree.cpp

```diff
@@ -237,8 +248,29 @@ CsgOpNode::CsgOpNode(std::vector<std::shared_ptr<CsgNode>> &&children,
   auto impl = impl_.GetGuard();
   impl->children_ = children;
   SetOp(op);
-  // opportunistically flatten the tree without costly evaluation
-  GetChildren(false);
+}
+
+std::shared_ptr<CsgNode> CsgOpNode::Boolean(
+    const std::shared_ptr<CsgNode> &second, OpType op) {
+  std::vector<std::shared_ptr<CsgNode>> children;
+
+  auto handleOperand = [&](const std::shared_ptr<CsgNode> &operand) {
+    if (auto opNode = std::dynamic_pointer_cast<CsgOpNode>(operand)) {
+      if ((opNode->IsOp(op) ||
+           (op == OpType::Subtract && opNode->IsOp(OpType::Add))) &&
```

Comment:
Now doing `(a-(b+c)-(d+e))` -> `(a-b-c-d-e)` here to play better w/ CsgOpNode::ToLeafNode (which turns it to `(a-batch(b+c+d+e))`